plugin/trino-pinot/src/test/java/io/trino/plugin/pinot/TestPinotSessionProperties.java

```diff
@@ -44,4 +48,21 @@ public void testConnectionTimeoutParsedProperly()
                 .build();
         assertThat(PinotSessionProperties.getConnectionTimeout(session)).isEqualTo(new Duration(0.25, TimeUnit.MINUTES));
     }
+
+    @Test
+    public void testQueryOptionsParsing()
+    {
+        PinotConfig config = new PinotConfig().setQueryOptions("enableNullHandling:true,skipUpsert:true,varcharOption:'value'");
+        PinotSessionProperties pinotSessionProperties = new PinotSessionProperties(config);
+        ConnectorSession session = TestingConnectorSession.builder()
+                .setPropertyMetadata(pinotSessionProperties.getSessionProperties())
+                .build();
+        String queryOptions = PinotSessionProperties.getQueryOptions(session);
+        assertThat(queryOptions).isEqualTo("enableNullHandling:true,skipUpsert:true,varcharOption:'value'");
+        Set<String> expected = Arrays.stream("SET enableNullHandling = true;\nSET skipUpsert = true;\nSET varcharOption = 'value';\n"
+                .split("\n")).collect(Collectors.toSet());
+        Set<String> options = Arrays.stream(PinotQueryBuilder.getQueryOptionsString(queryOptions).orElse("").split("\n"))
+                .collect(Collectors.toSet());
+        assertThat(options).isEqualTo(expected);
```

Comment:
Remove redundant `Arrays.stream` method:
```java
        assertThat(PinotQueryBuilder.getQueryOptionsString(queryOptions))
                .isEqualTo(Optional.of("""
                        SET enableNullHandling = true;
                        SET skipUpsert = true;
                        SET varcharOption = 'value';"""));
```

Reply:
The issue is that in map order is not consistent.

Reply:
The reason is you were using `HashMap`. The order should be consistent now.

Reply:
Reminder.